index.js

```diff
@@ -421,8 +421,9 @@ export default class Chk {
               res(buf)
             }
           }))
-      })
-    ))
+      }));
+    cb.directory = directory;
```

Comment:
This feels like a weird thing to do, and sort of assumes that `Tilesets._addFiles` is only ever called through this code location. This may be true currently, but it feels very brittle for future changes.

I think a better approach would be to pass in two functions: the one that is here now (`readCb`) and one that checks if a file exists (`existsCb`). Then you wouldn't need to try to pass the directory down, and all of the file operations would be located in one place.

Reply:
Additionally, since `Chk.customFileAccess` was intended to be a public API for cases where an user may want more control over file access/caching, the code should strive to be backwards compatible and handle cases where `existsCb` doesn't exist (Probably not trying to read the .vx4ex file in such case)